src/_pytest/mark/structures.py

```diff
@@ -93,20 +94,22 @@ def extract_from(cls, parameterset, legacy_force_tuple=False):
         if legacy_force_tuple:
             argval = (argval,)
 
-        if newmarks:
-            warnings.warn(MARK_PARAMETERSET_UNPACKING)
+        if newmarks and item is not None:
+            item.std_warn(MARK_PARAMETERSET_UNPACKING)
```

Comment:
item should be called `belonging_definition` and and as far as i can tell it **MUST** not be optional